crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -924,7 +931,7 @@ impl<'a> TsInterface<'a> {
                 }
                 self.src.push_str(";\n");
 
-                let variant_enum_name = format!("{}_variant", name).to_upper_camel_case();
+                let variant_enum_name = format!("{}_kind", name).to_upper_camel_case();
```

Comment:
I apologize for the mistake in #1166. I fixed it now. I think we should use `NodeType` to match the `.type` property we are adding below, not `NodeKind`.

I would also suggest using `type_enum_name` rather than `variant_enum_name` for the same reason.